### DISCUSSION TITLE
Parallel queries

### DISCUSSION THREAD

**Discussion by cevr** [urql-graphql/urql#2001] [Q&A] [answered]
Is there a way to fire multiple non-dependent queries in parallel?

**Comment by JoviDeCroock** [comment, marked as answer]
We'll need a bit more info, just using two `useQuery` hooks already do this

**Reply by cevr** [reply]
I've got two queries I use when I initialize the app. 
![image](https://user-images.githubusercontent.com/34170625/135534003-8aa0ab22-6b3d-4cac-a95a-6d9763c2fefc.png)


![image](https://user-images.githubusercontent.com/34170625/135533953-e745dd09-f8d9-4a1b-8d14-db4b18760c22.png)

However it looks like the calls happen sequentially.
Is there something I'm missing to make them run in parallel?

**Reply by JoviDeCroock** [reply]
normally they should already run in parallel, if you're using suspense, ... this could be that it's not the case 😅 could look into a codesandbox later tomorrow

**Reply by cevr** [reply]
Yeah I have suspense enabled by default

![image](https://user-images.githubusercontent.com/34170625/135535319-757df76d-8735-45c1-9228-022cd92c1e69.png)

**Reply by JoviDeCroock** [reply]
yeah, we can't do that much about that in that case 😅 grouping queries into one gql query would be a good way to circumvent that, you can try with suspense: false and you should see them in parallel. Essentially what suspense does is it throws the promise upwards so we hit query1 throw the promise, resolve it and go to query 2 here.

**Reply by cevr** [reply]
ah I see, yeah thats a good idea!

**Comment by FireworksX** [comment]
@JoviDeCroock Hi, how i can send multiply queries with suspense mode? `ReactQuery` has `useQueries` for wrapping few queries, urql has similar?

**Reply by JoviDeCroock** [reply]
GraphQL allows you to merge the `query-strings` together to one bigger query which should cover that case, unless there's something that makes that impossible?

**Reply by FireworksX** [reply]
I have old back-end, which can`t make Queries parallel from big query, and i wanna make parallel on Front-end
